src/Symfony/Bundle/FrameworkBundle/Templating/TemplateNameParser.php

```diff
@@ -50,11 +50,11 @@ public function parse($name)
         // normalize name
         $name = preg_replace('#/{2,}#', '/', str_replace('\\', '/', $name));
 
-        if (false !== strpos($name, '..')) {
+        if (str_contains($name, '..')) {
             throw new \RuntimeException(sprintf('Template name "%s" contains invalid characters.', $name));
         }
 
-        if (!preg_match('/^(?:([^:]*):([^:]*):)?(.+)\.([^\.]+)\.([^\.]+)$/', $name, $matches) || 0 === strpos($name, '@')) {
+        if (!preg_match('/^(?:([^:]*):([^:]*):)?(.+)\.([^\.]+)\.([^\.]+)$/', $name, $matches) || str_starts_with($name, '@')) {
```

Comment:
Not directly related and maybe micro optimisation, but wouldn't it be faster to do:
```suggestion
        if (str_starts_with($name, '@') || !preg_match('/^(?:([^:]*):([^:]*):)?(.+)\.([^\.]+)\.([^\.]+)$/', $name, $matches)) {
```

Reply:
Yes, probably. But I'd like to keep the change list straightforward and not include additional manual changes for now.

Reply:
Yes definitely not part of this PR 👍